static/app/components/events/autofix/useAutofixCodebaseIndexing.tsx

```diff
@@ -0,0 +1,95 @@
+import {useCallback, useEffect} from 'react';
+
+import {AutofixCodebaseIndexingStatus} from 'sentry/components/events/autofix/types';
+import {makeAutofixSetupQueryKey} from 'sentry/components/events/autofix/useAutofixSetup';
+import {
+  type ApiQueryKey,
+  setApiQueryData,
+  useApiQuery,
+  useQueryClient,
+} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePrevious from 'sentry/utils/usePrevious';
+import useProjectFromId from 'sentry/utils/useProjectFromId';
+
+const POLL_INTERVAL = 2500;
+
+const makeCodebaseIndexCreateUrl = (orgSlug: string, projectIdOrSlug: string): string =>
+  `/projects/${orgSlug}/${projectIdOrSlug}/autofix/codebase-index/create/`;
+
+const makeCodebaseIndexStatusQueryKey = (
+  orgSlug: string,
+  projectIdOrSlug: string
+): ApiQueryKey => [
+  `/projects/${orgSlug}/${projectIdOrSlug}/autofix/codebase-index/status/`,
+];
+
+const isPolling = (status: AutofixCodebaseIndexingStatus | undefined) =>
+  status === AutofixCodebaseIndexingStatus.INDEXING;
+
+export interface CodebaseIndexingStatusResponse {
+  status: AutofixCodebaseIndexingStatus;
+}
+
+export function useAutofixCodebaseIndexing({
+  projectId,
+  groupId,
+}: {
+  groupId: string;
+  projectId: string;
+}) {
+  const organization = useOrganization();
+  const projectSlug = useProjectFromId({project_id: projectId})?.slug;
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const {data: apiData} = useApiQuery<CodebaseIndexingStatusResponse>(
+    makeCodebaseIndexStatusQueryKey(organization.slug, projectSlug ?? ''),
+    {
+      staleTime: 0,
+      retry: false,
+      enabled: projectSlug !== undefined,
+      refetchInterval: data => {
+        if (isPolling(data?.[0]?.status)) {
+          return POLL_INTERVAL;
+        }
+        return false;
+      },
+    }
+  );
+
+  const status = apiData?.status ?? null;
+  const prevStatus = usePrevious(status);
+
+  const startIndexing = useCallback(() => {
+    if (!projectSlug) {
+      return;
+    }
+
+    // Triggering the create endpoint on the seer side should make it start return indexing status...
+    api.requestPromise(makeCodebaseIndexCreateUrl(organization.slug, projectSlug), {
+      method: 'POST',
+    });
```

Comment:
its a bit funky this promise is created and not caught or awaited but if you don't need to then go for it

Reply:
yeah that's true, but basically we want to have instant feedback on the other query key that it's indexing